src/Datadog.Trace.ClrProfiler.Managed.Loader/StartupLogger.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.IO;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Loader
+{
+    internal static class StartupLogger
+    {
+        private const string NixDefaultDirectory = "/var/log/datadog/dotnet";
+
+        private static readonly string LogDirectory = GetLogDirectory();
+        private static readonly string StartupLogFilePath = LogDirectory != null ? Path.Combine(LogDirectory, "dotnet-tracer-loader.log") : null;
+
+        public static void Log(string message, params object[] args)
+        {
+            try
+            {
+                if (StartupLogFilePath != null)
+                {
+                    try
+                    {
+                        using (var fileSink = new FileSink(StartupLogFilePath))
+                        {
+                            fileSink.Info($"[{DateTime.UtcNow}] {message}{Environment.NewLine}", args);
+                        }
+
+                        return;
+                    }
+                    catch
+                    {
+                        // ignore
+                    }
+                }
+
+                Console.WriteLine(message, args);
```

Comment:
This should probably be more intelligent, perhaps we just don't log at all at this point.

This change set already gives a lot more visibility than we have (none)

Reply:
Should this use `Console.Error` to write to stderr like you did in a different PR?